src/toil_scripts/gatk_germline/germline.py

```diff
@@ -490,81 +575,119 @@ def main():
 
-        # Parse config
-        config = {x.replace('-', '_'): y for x, y in yaml.load(open(options.config).read()).iteritems()}
-        config_fields = set(config)
-        required_fields = {'genome_fasta', 'output_dir', 'file_size', 'run_bwa', 'preprocess',
-                           'run_vqsr', 'joint', 'xmx'}
-        require(config_fields > required_fields,
-                'Missing config parameters:\n{}'.format(', '.join(required_fields - config_fields)))
-
-        config['file_size'] = human2bytes(config['file_size'])
-
-        if config['output_dir'] is None:
-            config['output_dir'] = options.output_dir \
+        # Parse inputs
+        inputs = {x.replace('-', '_'): y for x, y in
+                  yaml.load(open(options.config).read()).iteritems()}
+
+        required_fields = {'genome_fasta',
+                           'assembly',
+                           'output_dir',
+                           'run_bwa',
+                           'sorted',
+                           'preprocess',
+                           'preprocess_only',
+                           'run_vqsr',
+                           'joint',
+                           'run_oncotator',
+                           'cores',
+                           'file_size',
+                           'xmx',
+                           'suffix'}
+
+        input_fields = set(inputs.keys())
+        require(input_fields > required_fields,
+                'Missing config parameters:\n{}'.format(', '.join(required_fields - input_fields)))
+
+        if inputs['output_dir'] is None:
+            inputs['output_dir'] = options.output_dir \
                 if options.output_dir else os.path.join(os.getcwd(), 'output')
 
-        if config['suffix'] is None:
-            config['suffix'] = options.suffix if options.suffix else ''
-
-        if config['xmx'] is None:
-            pass
-        elif not config['xmx'][-1].isdigit():
-            config['xmx'] = human2bytes(config['xmx'])
-        else:
-            config['xmx'] = int(config['xmx'])
+        if inputs['suffix'] is None:
+            inputs['suffix'] = options.suffix if options.suffix else ''
 
-        if 'preprocess_only' not in config or config['preprocess_only'] is None:
-           config['preprocess_only'] = options.preprocess_only
+        if inputs['preprocess_only'] is None:
+            inputs['preprocess_only'] = options.preprocess_only
 
-        if config['run_vqsr']:
+        if inputs['run_vqsr']:
             vqsr_fields = {'phase', 'mills', 'dbsnp', 'hapmap', 'omni'}
-            require(config_fields > vqsr_fields,
+            require(input_fields > vqsr_fields,
                     'Missing parameters for VQSR:\n{}'
-                    .format(', '.join(vqsr_fields - config_fields)))
+                    .format(', '.join(vqsr_fields - input_fields)))
+
+        # Set resource parameters
+        inputs['xmx'] = convert_space_resource(inputs['xmx'], '10G')
```

Comment:
I don't know how I feel about having these parameters have default values. On one hand, that sounds useful for someone who wants to get up to speed ASAP. On the other hand, mistakenly falling back to the default seems like a great way to cause someone lots of pain. If you are going to provide defaults here, at least log that you're falling back on them.